cmd/launcher/launcher.go

```diff
@@ -243,18 +243,33 @@ func runLauncher(ctx context.Context, cancel func(), opts *options, logger log.L
 			return errors.Errorf("found no valid certs in PEM at path: %s", opts.rootPEM)
 		}
 	}
-
-	// connect to the grpc server
-	grpcConn, err := service.DialGRPC(opts.kolideServerURL, opts.insecureTLS, opts.insecureGRPC, opts.certPins, rootPool, logger)
-	if err != nil {
-		return errors.Wrap(err, "dialing grpc server")
-	}
-
 	// create a rungroup for all the actors we create to allow for easy start/stop
 	var runGroup run.Group
 
+	var client service.KolideService
+	{
+		switch opts.transport {
+		case "grpc":
+			grpcConn, err := service.DialGRPC(opts.kolideServerURL, opts.insecureTLS, opts.insecureGRPC, opts.certPins, rootPool, logger)
+			if err != nil {
+				errors.Wrap(err, "dialing grpc server")
+			}
+			defer grpcConn.Close()
+			client = service.NewGRPCClient(grpcConn, logger)
+			queryTargeter := createQueryTargetUpdater(logger, db, grpcConn)
+			runGroup.Add(queryTargeter.Execute, queryTargeter.Interrupt)
+		case "jsonrpc":
+			client = service.NewJSONRPCClient(opts.kolideServerURL, opts.insecureTLS, opts.insecureJSONRPC, opts.certPins, rootPool, logger)
+			if err != nil {
+				errors.Wrap(err, "create JSON RPC Client")
+			}
+		default:
+			errors.New("invalid transport option selected")
```

Comment:
needs a return

Reply:
oh good catch, also the err check on line 264 doesn't really check for an error in jsonrpc client creation.